### PR TITLE
feat: make aws and azure topic prefixes configurable in built-in bridge

### PR DESCRIPTION
**Pull request by jarhodes314**
## Proposed changes
Adds the ability to customise the bridge topic prefixes for AWS and Azure connections. Also improves the equivalent c8y configuration by supporting it in the mosquitto bridge and using the topic prefix in `te/.../c8y-mapper/...` topics used for operations.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
- #3011

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/c8y_mapper_ext/src/compatibility_adapter.rs [hunk 1]
```diff
@@ -140,16 +149,17 @@ fn convert_to_old_agent_request(
...
 ) -> Result<Option<MqttMessage>, String> {
     if let Ok(Value::Object(response)) = serde_json::from_slice(payload) {
         if let Some(Value::String(cmd_id)) = response.get("id") {
             // The new mapper expects command ids with a specific prefix
-            let topic_name = if cmd_id.starts_with("c8y-mapper") {
+            let topic_name = if cmd_id.starts_with(prefix) {
                 format!("te/device/main///cmd/{cmd_type}/{cmd_id}")
             } else {
-                format!("te/device/main///cmd/{cmd_type}/c8y-mapper-{cmd_id}")
+                format!("te/device/main///cmd/{cmd_type}/{prefix}-{cmd_id}")
```
Comment (Bravo555) [inline crates/extensions/c8y_mapper_ext/src/compatibility_adapter.rs:162]: question: so as of now we generate commands in this converter using a mapper topic prefix, but do we process (in `OperationHandler`) only commands matching the prefix?

Comment (jarhodes314) [reply crates/extensions/c8y_mapper_ext/src/compatibility_adapter.rs:162]: I've added a change to fix this

**Review by Bravo555 — COMMENTED** [review]
Looking good, but there's a merge commit 12d4267 that will probably have to get removed. Not sure if that's everything, when it is I'll give approval

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**